src/iSparrow/sparrow_watcher.py

```diff
@@ -0,0 +1,527 @@
+        # create a background watchertask such that the command is handed back to the parent process
+
+        self.may_do_work.set()
+        self.is_done_analyzing.clear()
+        self.watcher_process = multiprocessing.Process(target=watchertask, args=(self,))
+        self.watcher_process.daemon = True
+        self.watcher_process.name = "watcher_process"
+        self.watcher_process.start()
+
+    def restart(self):
+        """
+        restart Restart the watcher process. Must be called when, e.g., new models have been loaded or the input or output has changed.
+        """
+        self.stop()
+        self.start()
+
+    def pause(self):
+        """
+        pause Pause the watcher process.
+
+        Raises:
+            RuntimeError: When the watcher process is not running.
+        """
+        if self.watcher_process is not None and self.watcher_process.is_alive():
+            self.is_done_analyzing.wait()  # wait for the finish event
+            self.may_do_work.clear()
+        else:
+            raise RuntimeError("Cannot pause watcher process, is not alive anymore.")
+
+    def go_on(self):
+        """
+        go_on Continue the watcher thread.
+
+        Raises:
+            RuntimeError: When the watcher process is not running anymore
+        """
+        if self.watcher_process is not None and self.watcher_process.is_alive():
+            print("continue the watcher process")
+            self.may_do_work.set()
+        else:
+            raise RuntimeError("Cannot continue watcher process, is not alive anymore.")
+
+    def stop(self):
+        """
+        stop Stop the watcher process if it is still running.
+
+        Raises:
+            RuntimeError: When the watcher process is not running anymore
+        """
+        if self.watcher_process is not None and self.watcher_process.is_alive():
+            self.is_done_analyzing.wait()
+
+            print("stop the watcher process")
+            self.watcher_process.terminate()
+            self.watcher_process.join()
+            self.watcher_process.close()
+            self.watcher_process = None
+            self.may_do_work.clear()
+            self.is_done_analyzing.set()
+
+        else:
+            raise RuntimeError("Cannot stop watcher process, is not alive anymore.")
+
+    def clean_up(self):
+        """
+        clean_up Delete the input files which have been analyzed and delete them if 'delete_recordings' is 'on_cleanup'.
+                Files that have not yet been analyzed are analyzed *with the current model* before deletion. This means
+                that if they belong to a different batch that has been analyzed with a different model, they still will
+                be missing in there and be added to the current batch.
+        """
+
+        missings = []
+
+        if self.reanalyze_on_cleanup:
+            recording = self.set_up_recording()
+
+        for filename in self.input.iterdir():
+
+            condition = (
```

Comment:
make sure that the file that's currently under analysis is not taken into account upon cleanup if this is run concurrently.